### DISCUSSION TITLE
Laggy Hyprland with external monitor connected

### DISCUSSION THREAD

**Discussion by MaddPenguin** [hyprwm/Hyprland#4240] [Q&A]
I'm using Hyprland 0.33 with Nvidia (545.29.06)  and having this issue. My frame rate would dip to below 30Hz with external monitor connected and worsen with any movement on the screen. Without external monitor, 0.33 works flawless.

Version 0.32.3 works fine with external monitor connected. Anyone with similar problem or know the fix? Tq.

0.32.3
![0 32 3](https://github.com/hyprwm/Hyprland/assets/60058928/cad083b6-1d83-46d8-8cbd-17040f6b2a74)

0.33.1
![0 33](https://github.com/hyprwm/Hyprland/assets/60058928/2399b0e7-4fc1-4aac-8f02-febbdb893bdf)

**Comment by ChaseCooperr** [comment]
Same problem - driving me insane.

Any fix ?

**Comment by MaddPenguin** [comment]
Fixed by Vax in this issue #4252 

Install hyprland-git. Then add this in your config, assuming you are on Nvidia as well

```
opengl {
    nvidia_anti_flicker = false
}
```

**Reply by ChaseCooperr** [reply]
Thanks mate ! 

Actually got it working with the below.

https://wiki.hyprland.org/Configuring/Multi-GPU/

Thanks anyways

**Reply by shinobiultra** [reply]
This has for some reason caused insane input lag, especially noticeable when moving across workspaces. Mouse is like .5s behind, same goes for typing in terminal :/ Any hope of fixing this ?

Edit: made this a separate discussion #4575

**Reply by Vilen23** [reply]
How did you solve it....i am facing similar issue where external monitor is getting half the fps and talking about  https://wiki.hyprland.org/Configuring/Multi-GPU/ it says that it is by default set to integrated gpu so it shouldnt be a problem ig?
i am having a nvidia gpu
